app/src/main/java/org/cru/godtools/service/AppSeeAnalyticService.java

```diff
@@ -0,0 +1,128 @@
+package org.cru.godtools.service;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import com.appsee.Appsee;
+import com.appsee.AppseeListener;
+import com.appsee.AppseeScreenDetectedInfo;
+import com.appsee.AppseeSessionEndedInfo;
+import com.appsee.AppseeSessionEndingInfo;
+import com.appsee.AppseeSessionStartedInfo;
+import com.appsee.AppseeSessionStartingInfo;
+import com.crashlytics.android.Crashlytics;
+
+import org.cru.godtools.analytics.model.AnalyticsActionEvent;
+import org.cru.godtools.analytics.model.AnalyticsScreenEvent;
+import org.cru.godtools.analytics.model.AnalyticsSystem;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import java.util.Map;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+
+import static org.cru.godtools.analytics.model.AnalyticsScreenEvent.SITE_SECTION_MENU;
+import static org.cru.godtools.analytics.model.AnalyticsScreenEvent.SITE_SECTION_TOOLS;
+
+public class AppSeeAnalyticService implements Application.ActivityLifecycleCallbacks, AppseeListener {
+    @Nullable
+    private static AppSeeAnalyticService sInstance;
+
+    private AppSeeAnalyticService(Application application) {
+        EventBus.getDefault().register(this);
+        application.registerActivityLifecycleCallbacks(this);
+        Appsee.addAppseeListener(this);
+    }
+
+    @NonNull
+    public static synchronized AppSeeAnalyticService start(Application application) {
+        if (sInstance == null) {
+            sInstance = new AppSeeAnalyticService(application);
+        }
+        return sInstance;
+    }
+
+    @UiThread
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAnalyticScreenEvent(@NonNull final AnalyticsScreenEvent event) {
+        if (!event.isForSystem(AnalyticsSystem.APPSEE)) {
+            return;
+        }
+        // AppSee should match Adobe for integration
+        String siteSection = (SITE_SECTION_TOOLS.equals(event.getAdobeSiteSection()) ||
+                SITE_SECTION_MENU.equals(event.getAdobeSiteSection())) ? event.getAdobeSiteSection() :
+                event.getScreen();
+
+        String siteSubSection = event.getAdobeSiteSubSection();
+
+        String screen = siteSubSection != null ? String.format("%s - %s", siteSection, siteSubSection) : siteSection;
+
+        Appsee.startScreen(screen);
+    }
+
+    @Subscribe
+    public void onAnalyticActionEvent(@NonNull final AnalyticsActionEvent event) {
+        if (!event.isForSystem(AnalyticsSystem.APPSEE)) {
+            return;
```

Comment:
@frett I also noticed that the events are not listed as AppSee events.  Wasn't sure if that is correct or not.

Reply:
It is just the Pray question not the language toggle

Reply:
yeah, we will want to update those tools to trigger the analytics events for Appsee as well

Reply:
Is that something done in Android or the API?

Reply:
API